packages/appstore/src/components/appstore-banner/appstore-banner-image.tsx

```diff
@@ -2,20 +2,20 @@ import React from 'react';
 import SVGMigrateDesktopImage from 'Assets/svgs/banner/svg-migrate-desktop.svg';
 import SVGMigrateMobileImage from 'Assets/svgs/banner/svg-migrate-mobile.svg';
 
-export type BannerImageProps<T> = {
+export type TBannerImageProps<T> = {
     image: T;
     className?: string;
     width?: number;
 };
 
-type TAppstoreBannerImageList = keyof typeof AppstoreBannerImageList;
+type TAppstoreBannerImageListProps = keyof typeof AppstoreBannerImageList;
 
 const AppstoreBannerImageList = {
     svg_migrate_desktop: SVGMigrateDesktopImage,
     svg_migrate_mobile: SVGMigrateMobileImage,
 };
 
-const AppstoreBannerImage = ({ image, className, width }: BannerImageProps<TAppstoreBannerImageList>) => {
+const AppstoreBannerImage = ({ image, className, width }: TBannerImageProps<TAppstoreBannerImageListProps>) => {
     const Component = AppstoreBannerImageList[image] as React.ElementType;
```

Comment:
does typescript throw error if `as React.ElementType` is omitted?

Reply:
works fine